### PR TITLE
Execute non-data parallel tasks in parallel

### PR DESCRIPTION
**Pull request by dain**

### CODE REVIEW

**Comment by cberner** [comment]
I think you broke the tests. Make sure Travis passes before merging

**Comment by yuananf** [comment]
Can this PR improve join efficiency? If I increase experimental_task_parallel_count.

File: presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java [hunk 1]
```diff
@@ -19,6 +19,7 @@
     private static final String OPTIMIZE_HASH_GENERATION = "optimize_hash_generation";
     private static final String DISTRIBUTED_JOIN = "distributed_join";
     private static final String HASH_PARTITION_COUNT = "hash_partition_count";
+    private static final String TASK_PARALLEL_COUNT = "experimental_task_parallel_count";
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java]: I think we should just name it `task_parallel_count`. All these session properties are basically experimental.

**Comment by dain** [comment]
@yuananf this doesn't improve efficiency, it increases parallelism at the cost of some efficiency.  In the case of a join the overhead should be small, but for grouped aggregation it will be higher.

**Comment by yuananf** [comment]
I see, thanks!

**Comment by cberner** [comment]
Assigning back to you, since you mentioned it wasn't quite ready for review

**Comment by cberner** [comment]
Looks like there are still issues on Travis

**Comment by cberner** [comment]
I'd suggest trying this technique for projections and partial aggregations too. For expensive projections this could be very beneficial, as they may run single threaded if the user only is querying a single split

**Comment by cberner** [comment]
Looks good, but could you include some benchmark results?

**Pull request merged**